network/router/default_router.go

```diff
@@ -190,13 +167,186 @@ func (r *router) manageServiceRoutes(w registry.Watcher, metric int) error {
 	return watchErr
 }
 
+// watchTable watches routing table entries and either adds or deletes locally registered service to/from network registry
+// It returns error if the locally registered services either fails to be added/deleted to/from network registry.
+func (r *router) watchTable(w Watcher) error {
+	// wait in the background for the router to stop
+	// when the router stops, stop the watcher and exit
+	r.wg.Add(1)
+	go func() {
+		defer r.wg.Done()
+		<-r.exit
+		w.Stop()
+	}()
+
+	var watchErr error
+
+	for {
+		event, err := w.Next()
+		if err == ErrWatcherStopped {
+			break
+		}
+
+		if err != nil {
+			watchErr = err
+			break
+		}
+
+		u := &Update{
+			ID:        r.ID(),
+			Timestamp: time.Now(),
+			Event:     event,
+		}
+
+		select {
+		case <-r.exit:
+			return nil
+		case r.advertChan <- u:
```

Comment:
`advertChan` is a blocking channel; so this will block until update `u` is read by something
furthermore what happens if we close the `advertChan`? this will most likely `panic` as writing to closed channel causes `panic`